### DISCUSSION TITLE
Batteries not visible since update - Renogy via USB

### DISCUSSION THREAD

**Discussion by adamhratcliffe** [Louisvdw/dbus-serialbattery#699] [Q&A]
Hi,
I updated by driver to the latest version last night and now they do not show at all.  I have installed via SSH with no luck, USB with no luck. Tried to remove and reinstall but still no luck.

I have tried the troubleshooting steps with no luck either.
When I run 
tail -n 100 -f /data/log/serial-starter/current | grep dbus-serialbattery | tai64nlocal
It just hangs and I need to quit it.

When I run
bash /data/log/serial-starter/current
Here is the output

/data/log/serial-starter/current: line 1: @40000000648413341658fc44: command not found
/data/log/serial-starter/current: line 2: @40000000648413352964d704: command not found
/data/log/serial-starter/current: line 3: @400000006484133700bd2004: command not found
/data/log/serial-starter/current: line 4: @400000006484133a1628a92c: command not found
/data/log/serial-starter/current: line 5: @400000006484133d2a76c01c: command not found
/data/log/serial-starter/current: line 6: @400000006484133f01d10cbc: command not found
/data/log/serial-starter/current: line 7: @400000006484134014855584: command not found
/data/log/serial-starter/current: line 8: @4000000064841343295c7a64: command not found
/data/log/serial-starter/current: line 9: @400000006484134702b0a6ec: command not found
/data/log/serial-starter/current: line 10: @40000000648413481570b754: command not found
/data/log/serial-starter/current: line 11: @400000006484134928ba00a4: command not found
/data/log/serial-starter/current: line 12: @400000006484134d014d36bc: command not found
/data/log/serial-starter/current: line 13: @400000006484134e1408c6cc: command not found
/data/log/serial-starter/current: line 14: @400000006484134f270a7fcc: command not found
/data/log/serial-starter/current: line 15: @40000000648413523b5d1afc: command not found
/data/log/serial-starter/current: line 16: @400000006484135412fd03ec: command not found
/data/log/serial-starter/current: line 17: @400000006484135727f7ae8c: command not found
/data/log/serial-starter/current: line 18: @40000000648413583b518df4: command not found
/data/log/serial-starter/current: line 19: @400000006484135c145d0534: command not found
/data/log/serial-starter/current: line 20: @400000006484135d2742e54c: command not found
/data/log/serial-starter/current: line 21: @400000006484135e3a6a46ec: command not found
/data/log/serial-starter/current: line 22: @40000000648413621369bfdc: command not found
/data/log/serial-starter/current: line 23: @400000006484136326ec1e74: command not found
/data/log/serial-starter/current: line 24: @400000006484136439e73824: command not found
/data/log/serial-starter/current: line 25: @40000000648413681346595c: command not found
/data/log/serial-starter/current: line 26: @400000006484136b27b273e4: command not found
/data/log/serial-starter/current: line 27: @400000006484136c3a42e0fc: command not found
/data/log/serial-starter/current: line 28: @400000006484136e1154d9fc: command not found
/data/log/serial-starter/current: line 29: @4000000064841371262e680c: command not found
/data/log/serial-starter/current: line 30: @40000000648413743b509fac: command not found
/data/log/serial-starter/current: line 31: @400000006484137612bbf794: command not found
/data/log/serial-starter/current: line 32: @400000006484137725cc6e04: command not found
/data/log/serial-starter/current: line 33: @400000006484137a3a172994: command not found
/data/log/serial-starter/current: line 34: @400000006484137c11252af4: command not found
/data/log/serial-starter/current: line 35: @400000006484137d242c6234: command not found
/data/log/serial-starter/current: line 36: @4000000064841380387dad74: command not found
/data/log/serial-starter/current: line 37: @40000000648413820ffe46c4: command not found
/data/log/serial-starter/current: line 38: @400000006484138524aac994: command not found
/data/log/serial-starter/current: line 39: @400000006484138637d19e94: command not found
/data/log/serial-starter/current: line 40: @400000006484138a108827a4: command not found
/data/log/serial-starter/current: line 41: @400000006484138b238af9e4: command not found
/data/log/serial-starter/current: line 42: @400000006484138c369e911c: command not found
/data/log/serial-starter/current: line 43: @40000000648413900f8d0694: command not found
/data/log/serial-starter/current: line 44: @400000006484139122b7f65c: command not found
/data/log/serial-starter/current: line 45: @400000006484139235c849a4: command not found
/data/log/serial-starter/current: line 46: @40000000648413960f82f85c: command not found
/data/log/serial-starter/current: line 47: @400000006484139923f1d5ec: command not found
/data/log/serial-starter/current: line 48: @400000006484139a373097ec: command not found
/data/log/serial-starter/current: line 49: @400000006484139c0ecf8eac: command not found
/data/log/serial-starter/current: line 50: @400000006484139f23f3aaac: command not found
/data/log/serial-starter/current: line 51: @40000000648413a238d93af4: command not found
/data/log/serial-starter/current: line 52: @40000000648413a40fb96dc4: command not found
/data/log/serial-starter/current: line 53: @40000000648413a522e18ecc: command not found
/data/log/serial-starter/current: line 54: @40000000648413a83709f164: command not found
/data/log/serial-starter/current: line 55: @40000000648413aa0e1ae0c4: command not found
/data/log/serial-starter/current: line 56: @40000000648413ab21466ccc: command not found
/data/log/serial-starter/current: line 57: @40000000648413ae35860c9c: command not found
/data/log/serial-starter/current: line 58: @40000000648413b00db49544: command not found
/data/log/serial-starter/current: line 59: @40000000648413b322bc920c: command not found
/data/log/serial-starter/current: line 60: @40000000648413b43625abe4: command not found
/data/log/serial-starter/current: line 61: @40000000648413b80ef40e6c: command not found
/data/log/serial-starter/current: line 62: @40000000648413b9228ecb4c: command not found
/data/log/serial-starter/current: line 63: @40000000648413ba3594ca0c: command not found
/data/log/serial-starter/current: line 64: @40000000648413be0e6278e4: command not found
/data/log/serial-starter/current: line 65: @40000000648413bf215685e4: command not found
/data/log/serial-starter/current: line 66: @40000000648413c03481eec4: command not found
/data/log/serial-starter/current: line 67: @40000000648413c40e5e8914: command not found
/data/log/serial-starter/current: line 68: @40000000648413c722d1046c: command not found
/data/log/serial-starter/current: line 69: @40000000648413c83573f3cc: command not found
/data/log/serial-starter/current: line 70: @40000000648413ca0ce3a114: command not found
/data/log/serial-starter/current: line 71: @40000000648413cd21ee7084: command not found
/data/log/serial-starter/current: line 72: @40000000648413d036d76fb4: command not found
/data/log/serial-starter/current: line 73: @40000000648413d20e88738c: command not found
/data/log/serial-starter/current: line 74: @40000000648413d321e02074: command not found
/data/log/serial-starter/current: line 75: @40000000648413d6364bc9b4: command not found
/data/log/serial-starter/current: line 76: @40000000648413d80d929a0c: command not found
/data/log/serial-starter/current: line 77: @40000000648413d920a3a104: command not found
/data/log/serial-starter/current: line 78: @40000000648413dc34fc991c: command not found
/data/log/serial-starter/current: line 79: @40000000648413de0c4f7b4c: command not found
/data/log/serial-starter/current: line 80: @40000000648413e1212d7dfc: command not found
/data/log/serial-starter/current: line 81: @40000000648413e23487691c: command not found
/data/log/serial-starter/current: line 82: @40000000648413e60d345154: command not found
/data/log/serial-starter/current: line 83: @40000000648413e720063824: command not found
/data/log/serial-starter/current: line 84: @40000000648413e8331216cc: command not found
/data/log/serial-starter/current: line 85: @40000000648413ec0bc1162c: command not found
/data/log/serial-starter/current: line 86: @40000000648413ed1f154874: command not found
/data/log/serial-starter/current: line 87: @40000000648413ee325da37c: command not found
/data/log/serial-starter/current: line 88: @40000000648413f20c701ec4: command not found
/data/log/serial-starter/current: line 89: @40000000648413f5217d2884: command not found
/data/log/serial-starter/current: line 90: @40000000648413f634411534: command not found
/data/log/serial-starter/current: line 91: @40000000648413f80be54bb4: command not found
/data/log/serial-starter/current: line 92: @40000000648413fb20930704: command not found
/data/log/serial-starter/current: line 93: @40000000648413fe356c98fc: command not found
/data/log/serial-starter/current: line 94: @40000000648414000c98b94c: command not found
/data/log/serial-starter/current: line 95: @40000000648414011fe5a44c: command not found
/data/log/serial-starter/current: line 96: @40000000648414043409309c: command not found
/data/log/serial-starter/current: line 97: @40000000648414060b53509c: command not found
/data/log/serial-starter/current: line 98: @40000000648414071e350e1c: command not found
/data/log/serial-starter/current: line 99: @400000006484140a328cc9cc: command not found
/data/log/serial-starter/current: line 100: @400000006484140c09e3c6c4: command not found
/data/log/serial-starter/current: line 101: @400000006484140f1ef5e164: command not found
/data/log/serial-starter/current: line 102: @4000000064841410328422f4: command not found
/data/log/serial-starter/current: line 103: @40000000648414140bb9afa4: command not found
/data/log/serial-starter/current: line 104: @40000000648414151ef5878c: command not found
/data/log/serial-starter/current: line 105: @40000000648414163258bd94: command not found
/data/log/serial-starter/current: line 106: @400000006484141a0bd6d494: command not found
/data/log/serial-starter/current: line 107: @400000006484141b1edd7b4c: command not found
/data/log/serial-starter/current: line 108: @400000006484141c3208866c: command not found
/data/log/serial-starter/current: line 109: @40000000648414200bdaef5c: command not found
/data/log/serial-starter/current: line 110: @40000000648414232027b274: command not found
/data/log/serial-starter/current: line 111: @4000000064841424332d95dc: command not found
/data/log/serial-starter/current: line 112: @40000000648414260a6d70f4: command not found
/data/log/serial-starter/current: line 113: @40000000648414291f583d14: command not found
/data/log/serial-starter/current: line 114: @400000006484142c349e19b4: command not found
/data/log/serial-starter/current: line 115: @400000006484142e0c4716dc: command not found
/data/log/serial-starter/current: line 116: @400000006484142f1f91eab4: command not found
/data/log/serial-starter/current: line 117: @4000000064841432340bad54: command not found
/data/log/serial-starter/current: line 118: @40000000648414340b5e9b3c: command not found
/data/log/serial-starter/current: line 119: @40000000648414351e6ae744: command not found
/data/log/serial-starter/current: line 120: @40000000648414383285e814: command not found
/data/log/serial-starter/current: line 121: @400000006484143a0a63a13c: command not found
/data/log/serial-starter/current: line 122: @400000006484143d1f5a255c: command not found
/data/log/serial-starter/current: line 123: @400000006484143e328e2d44: command not found
/data/log/serial-starter/current: line 124: @40000000648414420b70d734: command not found
/data/log/serial-starter/current: line 125: @40000000648414431e2c5f74: command not found
/data/log/serial-starter/current: line 126: @4000000064841444315cae3c: command not found
/data/log/serial-starter/current: line 127: @40000000648414480a857d34: command not found
/data/log/serial-starter/current: line 128: @40000000648414491da63f84: command not found
/data/log/serial-starter/current: line 129: @400000006484144a3097d79c: command not found
/data/log/serial-starter/current: line 130: @400000006484144e0a193e44: command not found
/data/log/serial-starter/current: line 131: @40000000648414511e8c7134: command not found
/data/log/serial-starter/current: line 132: @400000006484145231527cdc: command not found
/data/log/serial-starter/current: line 133: @40000000648414540860ad1c: command not found
/data/log/serial-starter/current: line 134: @40000000648414571cf67f54: command not found
/data/log/serial-starter/current: line 135: @400000006484145a31d2b8fc: command not found
/data/log/serial-starter/current: line 136: @400000006484145c08fedd34: command not found
/data/log/serial-starter/current: line 137: @400000006484145d1c41b614: command not found
/data/log/serial-starter/current: line 138: @400000006484146030e20f9c: command not found
/data/log/serial-starter/current: line 139: @40000000648414620829707c: command not found
/data/log/serial-starter/current: line 140: @40000000648414631b04f43c: command not found
/data/log/serial-starter/current: line 141: @40000000648414662f5f5b84: command not found
/data/log/serial-starter/current: line 142: @400000006484146806e21f84: command not found
/data/log/serial-starter/current: line 143: @400000006484146b1c0dbd64: command not found
/data/log/serial-starter/current: line 144: @400000006484146c2f5e656c: command not found
/data/log/serial-starter/current: line 145: @4000000064841470085e150c: command not found
/data/log/serial-starter/current: line 146: @40000000648414711b704cb4: command not found
/data/log/serial-starter/current: line 147: @40000000648414722e85a13c: command not found
/data/log/serial-starter/current: line 148: @4000000064841476070a2d6c: command not found
/data/log/serial-starter/current: line 149: @400000006484147719f7452c: command not found
/data/log/serial-starter/current: line 150: @40000000648414782d421904: command not found
/data/log/serial-starter/current: line 151: @400000006484147c0719218c: command not found
/data/log/serial-starter/current: line 152: @400000006484147f1ba63194: command not found
/data/log/serial-starter/current: line 153: @40000000648414802e5fe12c: command not found
/data/log/serial-starter/current: line 154: @400000006484148205aa2454: command not found
/data/log/serial-starter/current: line 155: @40000000648414851a60961c: command not found
/data/log/serial-starter/current: line 156: @40000000648414882fbbead4: command not found
/data/log/serial-starter/current: line 157: @400000006484148a06dcd40c: command not found
/data/log/serial-starter/current: line 158: @400000006484148b19f7f8dc: command not found
/data/log/serial-starter/current: line 159: @400000006484148e2e369ac4: command not found
/data/log/serial-starter/current: line 160: @40000000648414900540bfdc: command not found
/data/log/serial-starter/current: line 161: @4000000064841491180256e4: command not found
/data/log/serial-starter/current: line 162: @40000000648414942c42f0b4: command not found
/data/log/serial-starter/current: line 163: @400000006484149603907bb4: command not found
/data/log/serial-starter/current: line 164: @400000006484149918535544: command not found
/data/log/serial-starter/current: line 165: @400000006484149a2b9bbc04: command not found
/data/log/serial-starter/current: line 166: @400000006484149e04614104: command not found
/data/log/serial-starter/current: line 167: @400000006484149f17136304: command not found
/data/log/serial-starter/current: line 168: @40000000648414a02a4b6674: command not found
/data/log/serial-starter/current: line 169: @40000000648414a403008ff4: command not found
/data/log/serial-starter/current: line 170: @40000000648414a516403c54: command not found
/data/log/serial-starter/current: line 171: @40000000648414a62982f5f4: command not found
/data/log/serial-starter/current: line 172: @40000000648414aa0306c9b4: command not found
/data/log/serial-starter/current: line 173: @40000000648414ad174d0104: command not found
/data/log/serial-starter/current: line 174: @40000000648414ae2a0656c4: command not found
/data/log/serial-starter/current: line 175: @40000000648414b001022014: command not found
/data/log/serial-starter/current: line 176: @40000000648414b315a66584: command not found
/data/log/serial-starter/current: line 177: @40000000648414b62a8d749c: command not found
/data/log/serial-starter/current: line 178: @40000000648414b801c42fc4: command not found
/data/log/serial-starter/current: line 179: @40000000648414b914f67a5c: command not found
/data/log/serial-starter/current: line 180: @40000000648414bc293dfdb4: command not found
/data/log/serial-starter/current: line 181: @40000000648414be0038e08c: command not found
/data/log/serial-starter/current: line 182: @40000000648414bf1360de6c: command not found
/data/log/serial-starter/current: line 183: @40000000648414c227d11b8c: command not found
/data/log/serial-starter/current: line 184: @40000000648414c33aaa2e4c: command not found
/data/log/serial-starter/current: line 185: @40000000648414c713fd06fc: command not found
/data/log/serial-starter/current: line 186: @40000000648414c8282d3994: command not found
/data/log/serial-starter/current: line 187: @40000000648414cc00e721ec: command not found
/data/log/serial-starter/current: line 188: @40000000648414cd13a361c4: command not found
/data/log/serial-starter/current: line 189: @40000000648414ce26e8a7bc: command not found
/data/log/serial-starter/current: line 190: @40000000648414d13b3c7784: command not found
/data/log/serial-starter/current: line 191: @40000000648414d3127e5a24: command not found
/data/log/serial-starter/current: line 192: @40000000648414d425801af4: command not found
/data/log/serial-starter/current: line 193: @40000000648414d73ad3a394: command not found
/data/log/serial-starter/current: line 194: @40000000648414db1478ce7c: command not found
/data/log/serial-starter/current: line 195: @40000000648414de294fdbec: command not found
/data/log/serial-starter/current: line 196: @40000000648414e20322e8ec: command not found
/data/log/serial-starter/current: line 197: @40000000648414e5181a0d34: command not found
/data/log/serial-starter/current: line 198: @40000000648414ea30ad5b6c: command not found
/data/log/serial-starter/current: line 199: @40000000648414ee0a4b71d4: command not found
/data/log/serial-starter/current: line 200: @40000000648414f11fa5ff54: command not found
/data/log/serial-starter/current: line 201: @40000000648414f83a84666c: command not found
/data/log/serial-starter/current: line 202: syntax error near unexpected token `('
/data/log/serial-starter/current: line 202: `@40000000648419ff12ec94e4 *** CCGX booted (0) ***'

Please can someone help. I am living in my van full time at the minute and really could do with monitoring. Thank you.

**Comment by adamhratcliffe** [comment]
when I try to reinstall the version I had before I get this error 

tar: conf/serial-starter.d: Cannot open: File exists
tar: Exiting with failure status due to previous errors

**Comment by ogurevich** [comment]
...just allowed me to ask the AI, the answer seems very plausible ...

It seems like you are dealing with a couple of issues here.

Firstly, regarding the issue with running `tail -n 100 -f /data/log/serial-starter/current | grep dbus-serialbattery | tai64nlocal` and it just hanging. The `tail -f` command is intended to keep the file open and show new lines as they are added. So, this command will keep running until you stop it (with Ctrl+C for example). This is normal behavior for `tail -f`. The reason you might not be seeing anything is that there might be no new lines containing "dbus-serialbattery" being added to the file while you are running this command.

The `tai64nlocal` command is used to convert TAI64N timestamps to a more human-readable format. If no output is being displayed, this may be because there are no lines with "dbus-serialbattery" to convert.

Regarding the issue with running `bash /data/log/serial-starter/current`, it seems the output you're seeing is because the lines in the file `/data/log/serial-starter/current` start with TAI64N timestamps, and bash is interpreting these as commands, which is not what you want.

TAI64N timestamps typically look like this: `@400000005f5e08922314f5ac`. When you run a file using `bash`, it treats the lines of the file as bash commands. Here, bash is trying to run each timestamp as a command, hence the "command not found" errors.

In order to read the contents of the file, you can use a command like `cat` or `less` instead of `bash`. For example:

```bash
cat /data/log/serial-starter/current
```

This will simply display the contents of the file.

If you want to convert the timestamps and read the file, you could use something like:

```bash
cat /data/log/serial-starter/current | tai64nlocal
```

This will convert the timestamps and display the contents of the file.

Please try these suggestions and see if it helps.

**Comment by mr-manuel** [comment]
Please check the updated FAQ for [downgrading](https://louisvdw.github.io/dbus-serialbattery/general/install/#downgrade-from--v100-to--v0143) and [troubleshooting](http://localhost:3000/dbus-serialbattery/troubleshoot/).

If you think that this a bug, please open an Issue -> [Bug report](https://github.com/Louisvdw/dbus-serialbattery/issues/new?assignees=&labels=bug&projects=&template=bug_report.yml).

**Comment by adamhratcliffe** [comment]
HI, thanks for your reply. I still cannot get the batteries to show. I am running Venus 3.0 installed using the latest version option and tried the troubleshooting but the battery still doesn't show.

I have tried to install this on a brand new Cerbo GX this morning and still doesn't show no matter which release I use. In the instructions, it mentions an inverter being connected - without an inverter via VESmart to USB will it not work?

**Comment by adamhratcliffe** [comment]
Hi. When trying to troubleshoot I get this error 

root@einstein:~# tail -F /data/log/dbus-serialbattery.ttyUSB0/current | tai64nlocal
2023-06-15 09:51:56.368862500 ERROR:SerialBattery:>>> ERROR: No reply - returning
2023-06-15 09:51:56.373659500 INFO:SerialBattery:Testing Renogy
2023-06-15 09:51:56.656132500 ERROR:SerialBattery:>>> ERROR: No reply - returning
2023-06-15 09:51:56.660649500 INFO:SerialBattery:Testing Renogy
2023-06-15 09:51:56.980719500 INFO:SerialBattery:RENOGY RBT100LFP12SH-G1
2023-06-15 09:51:57.805890500 ERROR:SerialBattery:Unexpected err=AttributeError("'Renogy' object has no attribute 'command_env_temp_count'"), type(err)=<class 'AttributeError'>
2023-06-15 09:51:57.806598500 INFO:SerialBattery:Testing Seplos
2023-06-15 09:51:58.830651500 ERROR:SerialBattery:>>> ERROR: No reply - returning
2023-06-15 09:51:59.332212500 ERROR:SerialBattery:ERROR >>> No battery connection at /dev/ttyUSB0
2023-06-15 09:51:59.332565500 

So it looks like it cannot detect the renogy because there is an error with the renogy line?

**Reply by mr-manuel** [reply]
Hi, the error is fixed in the dev branch. When running the installer please choose `nightly` and then `dev` branch.

Why did you open a duplicated discussion? https://github.com/Louisvdw/dbus-serialbattery/discussions/712

**Comment by adamhratcliffe** [comment]
Sorry I didn’t mean to open two. I’m living in my van at the minute so it’s stressful not being able to do things. Does the dev build reset upon restart?Sent from my iPhoneOn 15 Jun 2023, at 17:32, Manuel ***@***.***> wrote:﻿
Hi, the error is fixed in the dev branch. When running the installer please choose nightly and then dev branch.
Why did you open a duplicated discussion? #712

**Reply by mr-manuel** [reply]
> Does the dev build reset upon restart?

What do you mean exactly? The config is kept.
